src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions.cs

```diff
@@ -345,6 +345,12 @@ private static Option2<CodeStyleOption2<NamespaceDeclarationPreference>> CreateN
             "csharp_style_prefer_method_group_conversion",
             "TextEditor.CSharp.Specific.PreferMethodGroupConversion");
 
+        public static readonly Option2<CodeStyleOption2<bool>> PreferTrailingCommas = CreateOption(
+            CSharpCodeStyleOptionGroups.ExpressionLevelPreferences, nameof(PreferTrailingCommas),
+            defaultValue: s_trueWithSuggestionEnforcement,
+            "csharp_style_prefer_trailing_commas",
```

Comment:
i think it shoudl likely be `csharp_style_prefer_trailing_comma`.  @sharwell thoughts?  plural on singular here?  We don't say `prefer_switch_expressions` for example, just `prefer_switch_expression`.  And you can't have `x,,,`  and this doesn't impact anything but the last member (since the other members must have a comma already).  so it's really about a singular trailing comma in these lists.

--

i'm also somewhat on the fence here if we should say `csharp_style_prefer_trailing_enum_comma`.  The reason here is that we have proposals (i think i'm championing one) for things like a trailing comma in a parameter list.  i'm wary about having a single option now that some users might not want to apply to different constructs.

Reply:
@CyrusNajmabadi Per @sharwell's comment on the original issue:

>  it makes sense to add an editorconfig option to include this comma, which the various code generation features would then account for.

The "various code generation features" part implies that a single option is needed.

Reply:
reading through the critical decision was:

> @CyrusNajmabadi it seems like only one option would be needed here. I don't think StyleCop Analyzers ever had a request to split the single option in two, and even if it did it would be an extreme minority scenario.

I'm hesitantly ok with this.  However, if we have this, we need to update more than just the enum generator here.  We also need to check that object/collection-initializers and arrays are generated with this *and* checked for this.

Reply:
Also, switch/with expressions.

Reply:
@CyrusNajmabadi Can you point me to where object/collection initializers and switch/with expressions generators are?